engine/src/test/java/org/hibernate/validator/cfgtest/ConfigurationFilePropertiesTest.java

```diff
@@ -0,0 +1,150 @@
+		ValidatorFactory factory = hibernateConfig.buildValidatorFactory();
+		Validator validator = factory.getValidator();
+
+		ValidatorImpl hibernateValidatorImpl = (ValidatorImpl) validator;
+		BeanMetaDataManager bmdm = findPropertyOfType( hibernateValidatorImpl, BeanMetaDataManager.class );
+		MethodValidationConfiguration methodConfig = findPropertyOfType( bmdm, MethodValidationConfiguration.class );
+
+		Assert.assertTrue( methodConfig.isAllowMultipleCascadedValidationOnReturnValues() );
+	}
+
+	@Test
+	public void testAllowOverridingMethodAlterParameterConstraint() {
+		GenericBootstrap provider = Validation.byDefaultProvider();
+		Assert.assertNotNull( provider );
+
+		Configuration<?> config = provider.configure();
+		Assert.assertNotNull( config );
+		Assert.assertTrue( config instanceof HibernateValidatorConfiguration );
+
+		HibernateValidatorConfiguration hibernateConfig = (HibernateValidatorConfiguration) config;
+
+		// Note that the configuration from the XML is not read until the
+		// buildValidatorFactory() method is called.
+		ValidatorFactory factory = hibernateConfig.buildValidatorFactory();
+		Validator validator = factory.getValidator();
+
+		ValidatorImpl hibernateValidatorImpl = (ValidatorImpl) validator;
+		BeanMetaDataManager bmdm = findPropertyOfType( hibernateValidatorImpl, BeanMetaDataManager.class );
+		MethodValidationConfiguration methodConfig = findPropertyOfType( bmdm, MethodValidationConfiguration.class );
+
+		Assert.assertTrue( methodConfig.isAllowOverridingMethodAlterParameterConstraint() );
+	}
+
+	@Test
+	public void testAllowParallelMethodsDefineParameterConstraints() {
+		GenericBootstrap provider = Validation.byDefaultProvider();
+		Assert.assertNotNull( provider );
+
+		Configuration<?> config = provider.configure();
+		Assert.assertNotNull( config );
+		Assert.assertTrue( config instanceof HibernateValidatorConfiguration );
+
+		HibernateValidatorConfiguration hibernateConfig = (HibernateValidatorConfiguration) config;
+
+		// Note that the configuration from the XML is not read until the
+		// buildValidatorFactory() method is called.
+		ValidatorFactory factory = hibernateConfig.buildValidatorFactory();
+		Validator validator = factory.getValidator();
+
+		ValidatorImpl hibernateValidatorImpl = (ValidatorImpl) validator;
+		BeanMetaDataManager bmdm = findPropertyOfType( hibernateValidatorImpl, BeanMetaDataManager.class );
+		MethodValidationConfiguration methodConfig = findPropertyOfType( bmdm, MethodValidationConfiguration.class );
+
+		Assert.assertTrue( methodConfig.isAllowParallelMethodsDefineParameterConstraints() );
+	}
+
+	@After
+	public void afterTest() {
+
+	}
+
+	/**
+	 * Reflect into the subject and find the first property of the given type.
+	 *
+	 * @param subject - the instance to reflect on
+	 * @param clazz - exactly the class to match on
+	 *
+	 * @return
+	 */
+	private <T extends Object> T findPropertyOfType(Object subject, Class<T> clazz) {
```

Comment:
Uff, I guess I'd just expose the required member. All these types are internal anyways.